### PR TITLE
post and delete methods

### PR DESCRIPTION
**Pull request by Katerix**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Delete/DeleteRelatedFigureHandler.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private readonly IRepositoryWrapper _repositoryWrapper;
+
+    public DeleteRelatedFigureHandler(IRepositoryWrapper repositoryWrapper)
+    {
+        _repositoryWrapper = repositoryWrapper;
+    }
+
+    public async Task<Result<Unit>> Handle(DeleteRelatedFigureCommand request, CancellationToken cancellationToken)
+    {
+        var relation = await _repositoryWrapper.RelatedFigureRepository.GetFirstOrDefaultAsync(rel => rel.ObserverId == request.ObserverId && rel.TargetId == request.TargetId);
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Delete/DeleteRelatedFigureHandler.cs:18]: Could you move such a long LINQ line to the next line, like: 
....RelatedFigures
.GetFirst()...
?

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Сreate/CreateRelatedFigureHandler.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        if (entity1 is null)
+        {
+            return Result.Fail(new Error($"No existing streetcode with id: {request.ObserverId}"));
+        }
+
+        if (entity2 is null)
+        {
+            return Result.Fail(new Error($"No existing streetcode with id: {request.TargetId}"));
+        }
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Сreate/CreateRelatedFigureHandler.cs:34]: You should not reapet yourself. You could create one if condition like this:
if(entity1 is null || entity2 is null) {
...
}

Also names of entity1 and entity2 could be misunderstood. Maybe rename them as entityObserved and entityTarget.

File: Streetcode/Streetcode.WebApi/Controllers/Streetcode/RelatedFigureController.cs [hunk 3]
```diff
@@ -12,9 +16,15 @@ public async Task<IActionResult> GetByStreetcodeId([FromRoute] int streetcodeId)
         return HandleResult(await Mediator.Send(new GetRelatedFigureByStreetcodeIdQuery(streetcodeId)));
     }
 
-    [HttpGet("{tagId:int}")]
-    public async Task<IActionResult> GetByTagId([FromRoute] int tagId)
+    [HttpPost("{ObserverId:int}&{TargetId:int}")]
+    public async Task<IActionResult> Create([FromRoute] int ObserverId, int TargetId)
     {
-        return HandleResult(await Mediator.Send(new GetRelatedFiguresByTagIdQuery(tagId)));
+        return HandleResult(await Mediator.Send(new CreateRelatedFigureCommand(id1, id2)));
```
Comment (Kasterov) [inline Streetcode/Streetcode.WebApi/Controllers/Streetcode/RelatedFigureController.cs:22]: The same about naming... Please write idObserved and idTarget or else.

**Review by Kasterov — APPROVED** [review]
To wrap up, everything is good and should work :D!

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Delete/DeleteRelatedFigureHandler.cs (hunk 1 above)
Comment (Katerix) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Delete/DeleteRelatedFigureHandler.cs:18]: good idea! thanks!

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Сreate/CreateRelatedFigureHandler.cs (hunk 2 above)
Comment (Katerix) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/RelatedFigure/Сreate/CreateRelatedFigureHandler.cs:34]: i need to know which value is null to return a proper response

**Review by NadiaKishchuk — APPROVED** [review]

**Review by Tysyatsky — APPROVED** [review]

**Pull request merged**